processor/src/main/scala/it/pagopa/interop/authorizationmanagement/processor/key/KeyProcessor.scala

```diff
@@ -0,0 +1,45 @@
+package it.pagopa.interop.authorizationmanagement.processor.key
+
+import com.nimbusds.jose.jwk._
+import com.nimbusds.jose.util.X509CertUtils
+import com.nimbusds.jose.util.StandardCharset
+
+import java.util.Base64
+import scala.util.Try
+
+trait KeyProcessor {
+  def calculateKid(key: JWK): Either[Throwable, String]
+  def fromBase64encodedPEM(base64PEM: String): Either[Throwable, JWK]
+  def publicKeyOnly(key: JWK): Either[Throwable, Boolean]
+}
+
+object KeyProcessor extends KeyProcessor {
+
+  override def calculateKid(key: JWK): Either[Throwable, String] = Try {
+    key.computeThumbprint().toString
+  }.toEither.left.map(ex => new RuntimeException(ex.getLocalizedMessage))
+
+  override def fromBase64encodedPEM(base64PEM: String): Either[Throwable, JWK] = {
+    for {
+      decodedPem <- decodeBase64(base64PEM).toEither
+      key        <- fromPEM(decodedPem)
+    } yield key
+  }
+
+  private def decodeBase64(encoded: String): Try[String] = Try {
+    val decoded: Array[Byte] = Base64.getDecoder.decode(encoded.getBytes(StandardCharset.UTF_8))
+    new String(decoded, StandardCharset.UTF_8)
+  }
+
+  private def fromPEM(pem: String): Either[Throwable, JWK] = Try {
+    Option(X509CertUtils.parse(pem)) match {
+      case None    => Try { JWK.parseFromPEMEncodedObjects(pem) }.toEither
+      case Some(_) => Left[Throwable, JWK](new RuntimeException("The platform does not allow to upload certificates"))
```

Comment:
Can you type better this and following errors from `RuntimeException` to a specific error?